template-overlay.nix

```diff
@@ -0,0 +1,51 @@
+## File initially generated by createOverlay
+## and then supposedly modified manually.
+## Some hints for manual modifications are in the file,
+## but the full doc is on nixos / nix packages website:
+## https://nixos.org/manual/nixpkgs/stable/#sec-language-coq
+
+{ lib, mkCoqDerivation, which, coq
+  ## declare extra dependencies here, to be used in propagateBuildInputs e.g.
+  # , mathcomp, coq-elpi
+  , version ? null, shim ? false }:
+
+with lib; mkCoqDerivation {
+  pname = "template";
+  ## you can configure the domain, owner and repository, the default are:
+  # repo = "template";
+  # owner = "coq-community";
+  # domain = "github.com";
+
+  inherit version;
+  defaultVersion = with versions; switch coq.coq-version [
```

Comment:
We should clarify that this `defaultVersion` info is important for nixpkgs but will be ignored locally if this derivation corresponds to the main attribute or its version is overridden in `.nix/config.nix`.

Reply:
```suggestion
## The `defaultVersion` attribute is important for nixpkgs but can be kept unchanged
## for local usage since it will be ignored locally if
## - this derivation corresponds to the main attribute,
## - or its version is overridden (by a branch, PR, url or path) in `.nix/config.nix`.
  defaultVersion = with versions; switch coq.coq-version [
```

Reply:
@Zimmi48 WDYT?

Reply:
LGTM

Reply:
> EDIT: you seem to have an indentation issue though.

Where?